src/main/java/io/vertx/ext/amqp/impl/AmqpConnectionImpl.java

```diff
@@ -268,23 +271,62 @@ public AmqpConnection createReceiver(String address, AmqpReceiverOptions receive
     runWithTrampoline(v -> {
       ProtonReceiver receiver = connection.get().createReceiver(address, opts);
 
-      if (receiverOptions != null && receiverOptions.getQos() != null) {
-        receiver.setQoS(ProtonQoS.valueOf(receiverOptions.getQos().toUpperCase()));
+      if (receiverOptions != null) {
+        if (receiverOptions.getQos() != null) {
+          receiver.setQoS(ProtonQoS.valueOf(receiverOptions.getQos().toUpperCase()));
+        }
+
+        List<String> desired = receiverOptions.getDesiredCapabilities();
+        List<String> provided = receiverOptions.getCapabilities();
+
+        receiver.setDesiredCapabilities(desired.stream().map(Symbol::valueOf).toArray(Symbol[]::new));
```

Comment:
This is often not going to be what is desired. Most often the capabilities used are those applied to the remote terminus details (the source/target for receiver/sender respectively), and not the link (sender/receiver) itself.

I can think of a couple cases where link capabilities/properties are used in process of establishing support for certain functionality if it wasn't already established by connection-level capabilities/properties, but those aren't something applications typically do.